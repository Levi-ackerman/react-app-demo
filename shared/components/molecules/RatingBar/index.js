import React from 'react'
import PropTypes from 'prop-types'
import Rater from 'react-rater'
import styled from 'styled-components'
import { switchProp, prop } from 'styled-tools'
import 'react-rater/lib/react-rater.css'

const Wrap = styled.span`
  & .react-rater-star {
    color: #e0e0e0;
    font-size: ${switchProp(prop('variant'), {
      small: '0.8em',
      normal: '1.2em',
      large: '2em',
      larger: '2.4em',
    })};
  }
  .react-rater-star.is-active,
  .react-rater-star.will-be-active {
    color: #ffce00;
  }
`

const RatingBar = ({ variant, ...others }) => (
  <Wrap variant={variant}>
    <Rater {...others} />
  </Wrap>
)

RatingBar.propTypes = {
  variant: PropTypes.string,
}

RatingBar.defaultProps = {
  variant: 'normal',
}

export default RatingBar
