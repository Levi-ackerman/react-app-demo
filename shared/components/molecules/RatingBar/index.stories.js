import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import RatingBar from '.'

storiesOf('Rating', module)
  .add('default', () => <RatingBar />)
  .add('Size', () => (
    <div>
      <RatingBar variant="small" />
      <br />
      <RatingBar />
      <br />
      <RatingBar variant="large" />
      <br />
      <RatingBar variant="larger" />
      <br />
    </div>
  ))
