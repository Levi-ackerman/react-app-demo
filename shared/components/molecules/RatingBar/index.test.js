import React from 'react'
import renderer from 'react-test-renderer'
import RatingBar from '.'
import moment from 'moment'

describe('RatingBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RatingBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
