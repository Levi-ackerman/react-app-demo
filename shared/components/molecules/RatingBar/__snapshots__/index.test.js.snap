// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingBar renders correctly 1`] = `
<span
  className="RatingBar__Wrap-jjiBl GilFs"
>
  <div
    className="react-rater"
    onMouseLeave={[Function]}
  >
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="react-rater-star "
      >
        ★
      </div>
    </div>
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="react-rater-star "
      >
        ★
      </div>
    </div>
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="react-rater-star "
      >
        ★
      </div>
    </div>
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="react-rater-star "
      >
        ★
      </div>
    </div>
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="react-rater-star "
      >
        ★
      </div>
    </div>
  </div>
</span>
`;
